flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/DefaultExecutionGraphBuilder.java

```diff
@@ -143,7 +147,8 @@ public static DefaultExecutionGraph buildGraph(
                             executionJobVertexFactory,
                             jobGraph.getJobStatusHooks(),
                             markPartitionFinishedStrategy,
-                            nonFinishedHybridPartitionShouldBeUnknown);
+                            nonFinishedHybridPartitionShouldBeUnknown,
```

Comment:
Hey, @reswqa. I'm wondering since `nonFinishedHybridPartitionShouldBeUnknown` and `offloadShuffleDescriptorsThreshold` are both related to task deployment, could we merge them into a configuration class or a factory or something else? It could make the initialization of `ExecutionGraph` and other related components more readable. What's your opinion?

Reply:
Thanks for inviting me. I didn't carefully check the code, but I think they can be merged into one auxiliary class. If we plan to do this refactoring, other task deployment related fields (if any) should also be merged in.

Reply:
Thanks @Thesharing @reswqa. I created a new ticket (https://issues.apache.org/jira/browse/FLINK-32225) to do merge-related work to make this PR clean. I will finish that ticket first.